tests/crypto_policy_test.go

```diff
@@ -0,0 +1,240 @@
+			TLSProfileSpec: ocpv1.TLSProfileSpec{
+				Ciphers:       []string{"TLS_AES_128_GCM_SHA256", "TLS_CHACHA20_POLY1305_SHA256"},
+				MinTLSVersion: ocpv1.VersionTLS13,
+			},
+		},
+	}
+
+	tls10AllowedCipher = "ECDHE-ECDSA-AES128-SHA"
+
+	// Note that "crypto/tls" does not support setting max tls version to anything below 1.2
+	oldPermutation tlsConfigTestPermutation = tlsConfigTestPermutation{
+		openshiftTLSPolicy: &old,
+		allowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+		disallowedConfigs: []clientTLSOptions{},
+	}
+
+	intermediatePermutation tlsConfigTestPermutation = tlsConfigTestPermutation{
+		openshiftTLSPolicy: &intermediate,
+		allowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+		disallowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{tls10AllowedCipher},
+			},
+		},
+	}
+
+	modernPermutation tlsConfigTestPermutation = tlsConfigTestPermutation{
+		openshiftTLSPolicy: &modern,
+		allowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS13,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+		disallowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+	}
+
+	customPermutation tlsConfigTestPermutation = tlsConfigTestPermutation{
+		openshiftTLSPolicy: &custom,
+		allowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS13,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+		disallowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+	}
+)
+
+func getPemCertificate() []byte {
+	var apiService apiregv1.APIService
+	err := apiClient.Get(ctx, client.ObjectKey{Name: "v1.build.openshift.io"}, &apiService)
+	Expect(err).ToNot(HaveOccurred())
+	pemCertificate := apiService.Spec.CABundle
+	return pemCertificate
+}
+
+func tryToAccessEndpoint(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) (attemptedUrl string, err error) {
+	portInt, err := strconv.ParseUint(port, 10, 16)
+	Expect(err).ToNot(HaveOccurred())
+	conn, err := portForwarder.Connect(&pod, uint16(portInt))
+	Expect(err).ToNot(HaveOccurred())
+	defer conn.Close()
+
+	certPool := x509.NewCertPool()
+	pemCert := getPemCertificate()
+	certPool.AppendCertsFromPEM(pemCert)
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return conn, nil
+			},
+			TLSClientConfig: &tls.Config{CipherSuites: tlsConfig.CipherIDs(), MaxVersion: tlsConfig.MaxTLSVersion, RootCAs: certPool},
+		},
+	}
+
+	if subpath != "" {
+		subpath = "/" + subpath
+	}
+	attemptedUrl = fmt.Sprintf("https://ssp-webhook-service.kubevirt.svc:%s%s", port, subpath)
```

Comment:
What is the URL `ssp-webhook-service.kubevirt.svc` ? Can you maybe add a comment explaining it?